Content.Client/Paint/PaintVisualizerSystem.cs

```diff
@@ -0,0 +1,94 @@
+using Robust.Client.GameObjects;
+using static Robust.Client.GameObjects.SpriteComponent;
+using Content.Shared.Clothing;
+using Content.Shared.Hands;
+using Content.Shared.Paint;
+using Robust.Client.Graphics;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.Paint;
+
+public sealed class PaintedVisualizerSystem : VisualizerSystem<PaintedComponent>
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly IPrototypeManager _protoMan = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PaintedComponent, HeldVisualsUpdatedEvent>(OnHeldVisualsUpdated);
+        SubscribeLocalEvent<PaintedComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<PaintedComponent, EquipmentVisualsUpdatedEvent>(OnEquipmentVisualsUpdated);
+    }
+
+
+    protected override void OnAppearanceChange(EntityUid uid, PaintedComponent component, ref AppearanceChangeEvent args)
+    {
+        var shader = _protoMan.Index<ShaderPrototype>(component.ShaderName).Instance();
+
+        if (args.Sprite == null)
+            return;
+
+        // What is this even doing? It's not even checking what the value is.
+        if (!_appearance.TryGetData(uid, PaintVisuals.Painted, out bool isPainted))
+            return;
```

Comment:
```suggestion
        var shader = _protoMan.Index<ShaderPrototype>(component.ShaderName).Instance();

        if (args.Sprite == null)
            return;

        // What is this even doing? It's not even checking what the value is.
        if (!_appearance.TryGetData(uid, PaintVisuals.Painted, out bool isPainted))
            return;
```
I'm reasonably sure that _protoMan has a TryIndex that can be used as a guard clause here, and thus all 3 of these can be combined.